neo/rawio/alphaomegarawio.py

```diff
@@ -0,0 +1,1252 @@
+        ("Digital Port", "InPort", "InPort"),
+        ("Internal Detection", "Internal Detection", "Internal Detection"),
+    )
+
+    def __init__(self, dirname="", prune_channels=True):
+        super().__init__(dirname=dirname)
+        self.dirname = Path(dirname)
+        self._filenames = {}
+        if self.dirname.is_dir():
+            self._explore_folder()
+        else:
+            self.logger.error(f"{self.dirname} is not a folder")
+        self._prune_channels = prune_channels
+        self._opened_files = {}
+        self._ignore_unknown_blocks = True  # internal debug property
+
+    def _explore_folder(self):
+        """
+        Explores an AlphaOmega folder and tries to load the index files (*.lsx)
+        to get the data files (*.mpx). If the folder contains only *.mpx files,
+        it will load them without splitting them into blocks.
+        It does not explores the folder recursively.
+        """
+        self._filenames = defaultdict(list)
+
+        # first check if there is any *.mpx files
+        filenames = list(filter(lambda x: x.is_file(), self.dirname.glob("*.mpx")))
+        if not filenames:
+            self.logger.error(f"Found no AlphaOmega *.mpx files in {self.dirname}")
+        else:
+            index_files = list(
+                filter(lambda x: x.is_file(), self.dirname.glob("*.lsx"))
+            )
+            # the following is not very usefull but we should expect the original
+            # names to be in increasing lexical order
+            index_files.sort()
+            if not index_files:
+                self.logger.warning(
+                    "No *.lsx files found. Will try to load all *.mpx files"
+                )
+                self._filenames[""].extend(filenames)
+            else:
+                for index_file in index_files:
+                    with open(index_file, "r") as f:
+                        for line in f:
+                            # a line is a Microsoft Windows path. As we cannot
+                            # instanciate a WindowsPath on other OS than MS
+                            # Windows, we use the PureWindowsPath class
+                            filename = PureWindowsPath(line.strip())
+                            filename = self.dirname / filename.name
+                            if not filename.is_file():
+                                self.logger.warning(f"File {filename} does not exist")
+                            else:
+                                self._filenames[index_file.name].append(filename)
+                                filenames.remove(filename)
+                if filenames:
+                    self.logger.info(
+                        "Some *.mpx files not referenced. Will try to load them."
+                    )
+                    self._filenames[""].extend(filenames)
+
+    def _source_name(self):
+        return str(self.dirname)
+
+    def _read_file_blocks(self, filename, prune_channels=True):
```

Comment:
To avoid any confusion with neo.Block I would use "datablock"' instead of block.